### DISCUSSION TITLE
Custom javascript with page conditional

### DISCUSSION THREAD

**Discussion by brunushky** [holger1411/understrap#1879] [Q&A] [answered]
Hi, I would like to know if is possible to add a script in custom-javascript.js, but made part of it enable only for specific page???

I mean I understand that all the script in custom-javascript.js is compiled, but there is any option to get this behavior or should I just create two separate scripts and enable or disable one of them in the function.php file, like in any other theme? 

And in that case, how can I add this custom scripts to auto minify on npm run watch? 

Thanks.

**Comment by bacoords** [comment]
You could check for a class in the `body` element before running the function. Basically every page, post, etc in WordPress has different combinations of unique classes.

**Reply by brunushky** [reply]
Ok, I was thinking this could be a practical solution at least. 

Thanks for you fast reply!!!.

**Comment by IanDelMar** [comment, marked as answer]
Depending on the size of the custom JS I'd suggest to load a separate JS file. There is no need to load the JS on pages you know it won't be executed.

To output a separate file you can place the let's say another.js in src/js and change your rollup.config.js like
```diff
-module.exports = {
+const jsFile1 = {
  input: [path.resolve(__dirname, '../js/bootstrap.js'), path.resolve(__dirname, '../js/skip-link-focus-fix.js'), path.resolve(__dirname, '../js/custom-javascript.js')],
  output: {
    banner,
    file: path.resolve(__dirname, `../../js/${fileDest}`),
    format: 'umd',
    globals,
    name: 'understrap'
  },
  external,
  plugins
}
+const jsFile2 = {
+  input: path.resolve(__dirname, '../js/another.js'),
+  output: {
+    banner,
+    file: path.resolve(__dirname, `../../js/another.js`),
+    format: 'umd',
+    globals,
+    name: 'another'
+  },
+  external,
+  plugins
+}
+
+module.exports = [ jsFile1, jsFile2 ];
```
In your package.json you have to add the minify task:
`"js-minify-another": "terser  --config-file src/build/terser.config.json --output js/another.min.js js/another.js",` and
add the new task to the `js` task:
```diff
- "js": "npm-run-all js-compile js-minify",
+ "js": "npm-run-all js-compile js-minify js-minify-another",
```

**Reply by brunushky** [reply]
Great!!! I was looking for something like this....   I did found the **rollup.config.js** option settings, but was stuck with the terser minify part. 

Just one question about the sourceMap option in **terser.config.json**, this part isn't clear for me yet. 

`"sourceMap": {
        "filename": "js/child-theme.js.map",
        "url": "child-theme.min.js.map"
    }`

Thanks.

**Reply by IanDelMar** [reply]
Actually, I recommend using the terser.config.json and the js-minify task from the develop branch.

Terser config:
https://github.com/understrap/understrap/blob/3a47cd078daae3e04be899c89aea3fb88bf777c8/src/build/terser.config.json

task in package.json:
```diff
-"js-minify": "terser  --config-file src/build/terser.config.json --output js/theme.min.js js/theme.js",
+"js-minify": "terser js/theme.min.js --config-file src/build/terser.config.json --source-map \"content=js/theme.js.map,url=theme.min.js.map,filename=theme.min.js\" --output js/theme.min.js",`
+"js-minify-another": "terser js/another.min.js --config-file src/build/terser.config.json --source-map \"content=js/another.js.map,url=another.min.js.map,filename=another.min.js\" --output js/another.min.js",`
```

There are issues with the souremap in the current build process (Understrap 1.1.0) - see #1641 - which should have been fixed in #1632.

**Reply by brunushky** [reply]
Oh, great I will try it!!! Thanks!!!
